### DISCUSSION TITLE
Status message

### DISCUSSION THREAD

**Discussion by ollybeal** [UKHomeOffice/design-system#419] [General]
Pattern (https://design.homeoffice.gov.uk/components?name=Status%20message) is being developed for an internal service where multiple users can see the same case - pattern is shown near the top of the screen. Used to show when ownership changes and read only nature depending on which users views the case. Allows option for users to take over ownership.

In research users saw and understood what the message meant and how to interact with it to perform their tasks.

<img width="687" alt="Screenshot 2022-07-22 at 11 31 33" src="https://user-images.githubusercontent.com/19707819/180421354-4c4c5f29-dbc8-4ff8-a4d8-a0177d8a38e4.png">
.

**Comment by tomyems** [comment]
Couple of questions, how many participants did the research have?

I propose to update the Research part of the [Status message](https://design.homeoffice.gov.uk/components?name=Status%20message) component with:

## Research

For an internal service, this was used at the top of the screen of a case management system to:
- show who owns the case
- whether the case is read only
- allow users to take over the case

Research showed that users understood the message and how to interact with it.

**Comment by ollybeal** [comment]
Thanks @tomyems - the research was conducted with six users of the service.

**Comment by tomyems** [comment]
As an iteration:

For an internal service, this was used at the top of the screen of a case management system to:

- show who owns the case
- whether the case is read only
- allow users to take over the case

(Show example as a story, so with the HTML and React code for people to re-use)

Research with 6 users showed that users understood the message and how to interact with it.

**Reply by tomyems** [reply]
We'll also add a Contribute section like we have for the [Date-time component](https://design.homeoffice.gov.uk/components?name=Date-time) with a link to this discussion.

**Comment by tomyems** [comment]
The proposal is something like this https://62f3df93fa4c5d0d2b9cd63e--hods.netlify.app/components?name=Status%20message

(To see full set of changes see: https://github.com/UKHomeOffice/design-system/pull/444)

**Reply by tomyems** [reply]
@ollybeal - your example is actually quite hard to replicate in the component, and shows that we might need to update and improve the component that we have :)

**Comment by tomyems** [comment]
From a content 2i, added verb 'show' to make the bullets consistent. So it now looks like:

For an internal service, this was used at the top of the screen of a case management system to:

- show who owns the case
- show whether the case is read only
- allow users to take over the case
(Show example as a story, so with the HTML and React code for people to re-use)

Research with 6 users showed that users understood the message and how to interact with it.

**Reply by Sarraounia** [reply]
@tomyems couple of comments from me about the content. 

1. The lead-in line read to me like it describes the service twice currently. How about the below?

> For an internal case management service, this was used at the top of the screen to:

2. In the second bullet point, replace 'whether' with 'if' for simplicity.

> show if the case is read only

Also, had a thought if we need a 'Writing for this component' section for some recommendations around what content goes into status messages in different scenarios? Might be one to pick up separately though!

**Reply by tomyems** [reply]
Ah my mistake, forgot to update this thread. This is the latest version: https://62f4c6a6324ac21a19b7d268--hods.netlify.app/components?name=Status%20message.

I'll change the 'whether' to 'if', and after a conversation with Sarah on Slack, have described the system in a slightly different way:

> For an service to assess border risk, this was used at the top of the screen to:

(just spotted the incorrect 'an')

**Reply by tomyems** [reply]
And yes it probably should have a writing for this component section as well :)

**Reply by Sarraounia** [reply]
All sounds good - thanks @tomyems

**Reply by tomyems** [reply]
Updated 'whether' to 'if': https://6304ccddfb22d77dee029268--hods.netlify.app/components

**Comment by tomyems** [comment]
@ollybeal This research example is now published on the Home Office design system: https://design.homeoffice.gov.uk/components?name=Status%20message. - Thanks for contributing :)

**Comment by tomyems** [comment]
Proposal to add some guidance for writing status messages - improvements and suggestions welcome:

> ### Writing for this component
> The status message should be as concise as possible. The bold part and button should be short, 1 or 2 words. 
> 
> If text in the middle is necessary, then ideally 1 line. It does not need to be in full sentences and does not need to end in a full stop. It should be plain text with no formatting. 
> 
> Any buttons should be verbs. 
> 
> Do not use links, they do not contrast well with the background, use buttons instead. 
> 
> If it’s about people, make sure your content is dynamic and personalised. Use ‘person’ when it’s about one person and ‘people’ for more than one and include names if you can.

**Comment by tomyems** [comment]
Proposing to take to to the working group next Friday, 21 October. If you could provide comments by then.

> ### When to use this component 
> 
> Use status messages to: 
> 
> - notify the user of a change 
> - tell the user something important 
> - inform the user that they need to do something 
> 
> They are like alerts but are primarily used for internal services and can include actions for users to take. For less prominent status indicators you could use [tags](https://design-system.service.gov.uk/components/tag/). 
> 
> Limit to important information, or when it’s changed. Overuse risks the message getting lost. 
> 
> ### Writing for this component 
> 
> The status message should be as concise as possible and relate to the whole page. The bold part and button should be short, 1 or 2 words. 
> 
> Any buttons should be verbs. 
> 
> If text in the middle is necessary, then ideally 1 line. It does not need to be in full sentences and does not need to end in a full stop. It should be plain text with no formatting. 
> 
> Do not use links, they do not contrast well with the background, use buttons instead. 
> 
> If it’s about people, make sure your content is dynamic and personalised. Use ‘person’ when it’s about one person and ‘people’ for more than one and include names if you can.

**Reply by tomyems** [reply]
View a draft: https://63524eaf8f87b55bd6bc4f67--hods.netlify.app/components?name=Status%20message

**Comment by tomyems** [comment]
Reviewed by design system working group. To do:

- clarify 'make sure your content is dynamic and personalised' to mean include person's name in the status message.

**Comment by tomyems** [comment]
Updated to allow design as described https://design.homeoffice.gov.uk/components?name=Status%20message with 2 lines of text.

**Comment by benjystanton** [comment]
Would it be possible to enable flex-wrap on the links please? This would allow the links to wrap over multiple lines when there is not enough horizontal space. 

Lack of horizontal space can be caused…
- if the viewer is using a medium viewort (but still wide enough to display the desktop layout)
- if multiple buttons are added
- if longer text is used

Screenshot of problem
<img width="466" alt="A blue box containing white buttons, the white buttons are too wide and break out of the container on the right hand side" src="https://github.com/UKHomeOffice/design-system/assets/2545065/ff510c81-18f3-4edc-bd6e-d48f2420b0eb">


Screenshot of suggested solution
<img width="461" alt="A blue box containing white buttons, the white buttons wrap onto 2 lines, making sure they always fit inside the container" src="https://github.com/UKHomeOffice/design-system/assets/2545065/ecd393e4-eccd-46fc-84f4-3e5f3a6905f0">

**Comment by willrichards** [comment]
@benjystanton we discussed this today at the design system working group. If we can put a pull request together, we can get it reviewed by the development team and hopefully added. Happy to support with this.

**Comment by daniel-ac-martin** [comment]
(For posterity.)

This seems more likely with 3 or more buttons.

Placing `flex-wrap: wrap` on the flex container (the `<ul>`) stops the issue as presented, but does prevent wrapping within the button text. i.e. Taller buttons.

It's not currently obvious to me why the `<ul>` is overflowing in the first place. (Maybe we can just prevent that?)

I wonder if there might be a better solution. e.g. Should we be using a 'button group' instead? (And would it solve the problem if we did.)

**Reply by daniel-ac-martin** [reply]
Update: A button group does seem to fix this. I'm not exactly sure how. (Perhaps by preventing the overflow of the container?)

**Reply by daniel-ac-martin** [reply]
See: https://github.com/UKHomeOffice/design-system/issues/572

**Comment by benjystanton** [comment]
Thanks both, I'll have a play with button group in my prototype too.

**Comment by benjystanton** [comment]
Button group looks like it does the trick, thanks. But, in my original example above, I also reduced the margin on the links themselves. Otherwise when the list begins to wrap the gap between the lines gets quite big. See screenshot. Just mentioning in case we can address this at the same time.

![Screenshot of blue banner containing text and 2 rows of links](https://github.com/UKHomeOffice/design-system/assets/2545065/5eac670a-0669-4339-a294-af1b10ae779c)

**Reply by benjystanton** [reply]
Adjusting the margin to something like the following helps (although there may be better ways to achieve this).
```
.hods-status-message__actions__link {
  margin-top: 4px;
  margin-bottom: 4px;
  }
```

**Reply by andymarshall-35** [reply]
Where we're using `display: flex; `we can also use `gap: 10px 5px;` to apply space around the flex element's contained buttons - gap works a lot neater than margin, as it only applies itself when it needs it to space the buttons from other neighbouring buttons (unlike margin that is always applied to the button, even if there are no other buttons to push away from, leading to other inconsistencies).

**Reply by benjystanton** [reply]
Thanks Andy, I didn't know this!

**Comment by daniel-ac-martin** [comment]
Suggested fix is here: #574

It does change some of the vertical spacing though, so it's worth someone reviewing the visual changes in Chromatic.

You'll need to use your browser's dev tools to add in some extra buttons and then narrow your browser width to see the original issue.

**Comment by benjystanton** [comment]
We discussed this component at the Design System Working Group today.

We think the open pull request is ready to go https://github.com/UKHomeOffice/design-system/pull/574

A question was raised over the accessibility of the component in general. I'll email the HO accessibility assurance team and ask them to add a review of the component to their backlog. It looks as if the semantics could be improved at least. For eaxmple, the component has what looks like a heading, but it's just a div.

**Comment by knajdek** [comment]
We reviewed accessibility of the component and found the following issues:

Use of white text and the blue background attracts sighted users’ attention and conveys importance of the message. This is not communicated to screen reader users via semantic markup. 
 
In addition, relationship between information within the status message is communicated via use of bold text and normal size text, this is also not communicated to screen reader users.
 
Bold text within status message should be marked up as a heading as it looks like one. This will help screen reader users easily navigate to the status message and understand all information within it. 

In addition, if the status message dynamically appears on the page, for example as a result of user action, this also needs to be announced to screen reader users. Consider using [ARIA live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) for this.

Focus indicator on buttons might be difficult to see for users with colour vision deficiency. It relies on colour change from white to yellow and a very subtle change to the border appearance that is very easy to miss.

**Reply by benjystanton** [reply]
Thanks! As mentioned I'd be happy to have a go at improving my team's prototype to include some of these accessibility improvements, then I could bring that back to the working group to check it meets the criteria.

**Comment by benjystanton** [comment]
I've started tinkering with some of @knajdek's (and other's) feedback on the status message banner to try and improve accessibility. Just working on my team's private prototype for now.

Here's a a screenshot for context (although I'm not proposing any visual changes).

<img width="955" alt="Blue banner with white text and 3 buttons. The text says that the task needs review, who's working on it and when it's due" src="https://github.com/user-attachments/assets/f8cb42f8-a6eb-4661-8ada-97e48a24e821">

Done…

- Change main status text ("Needs review") from `div` to `h2`
- Removed extra div that wrapped the "Needs review" and 3 lines of text (don't think it was needed?)
- Add `role="status" aria-live="polite"` to main container
- Add `hods-date-time` component to review date
- Changed the 3 lines of a text from a single `<p>` with `<br>`s to a `<dl>`

Code…

```
<div class="hods-status-message" role="status" aria-live="polite">
            <h2 class="hods-status-message__status">Needs review</h2>
            <div class="hods-status-message__extra">
                <dl>
                    <div>
                        <dt>Assigned to </dt>
                        <dd>North Team</dd>
                    </div>
                    <div>
                        <dt>Claimed by </dt>
                        <dd>Kelvin Phillips</dd>
                    </div>
                    <div>
                        <dt>Review by </dt>
                        <dd><time class="hods-date-time" datetime="2024-08-31T19:00:00.000Z">31 August 2024, 19:00</time></dd>
                    </div>
                </dl>
            </div>
            <ul class="hods-status-message__actions">
                <li class="hods-status-message__actions__item hods-status-message__actions__item--active"><a class="hods-status-message__actions__link" href="#">Edit task</a></li>
                <li class="hods-status-message__actions__item hods-status-message__actions__item--active"><a class="hods-status-message__actions__link" href="#">Assign task</a></li>
                <li class="hods-status-message__actions__item hods-status-message__actions__item--active"><a class="hods-status-message__actions__link" href="#">Close task</a></li>
            </ul>
        </div>
```

Feedback welcome!

**Comment by benjystanton** [comment]
Working on some changes to the component here https://github.com/UKHomeOffice/design-system/issues/607

**Comment by mgifford** [comment]
What is the status of this?

**Comment by millie-drs** [comment]
We are using the status pattern but with slight amendments. 
-Context under the blue banner (not within)
-Ability to add notes 
We found that the status message sometimes needed more context than would fit in the blue area. Instead of trying to fit white text into the blue area, we decided to combine it with the pattern used for the GDS ‘notification banner’ - ie having black body text on white. 

We also found that where a status has been changed, someone else picking it up may need to be provided with context on why it’s been changed. So, we’ve added a notes field also.

One thing that initial research has found is that the blue status banner without buttons was skipped over and not noticed by some users. We have a hypothesis that this may be affected by the new branding because the status banner, now being the same colour, was mistaken for the branded header area and ignored.

Research findings seem to indicate that having the larger area for the context helped with people not skipping over and ignoring. 
We are continuing to test but sharing our early findings here.
<img width="993" height="754" alt="Screenshot 2026-06-24 at 13 04 20" src="https://github.com/user-attachments/assets/07035f2d-6431-4e05-b757-0dbb5ce70020" />
